File: c_work21/20210127/ex02.c
#include <stdio.h>

void main()
{
    int arr[5] = {0, 1, 2, 3, 4};
    int b = 20;

    printf("arr = %d\n", arr);
    printf("&arr[0] = %d\n", &arr[0]);
    printf("&arr[1] = %d\n", &arr[1]);
    printf("&arr[2] = %d\n", &arr[2]);
    printf("&arr[3] = %d\n", &arr[3]);
    printf("&arr[4] = %d\n", &arr[4]);

    //arr = &b; //arr의 포인터변수값은 상수이기 때문에 변경이 되지 않는다 !
}

File: java/20210315/src/Ex02.java
import java.util.Scanner;

public class Ex02 {
	Ex02(){
		Scanner scan = new Scanner(System.in);
		System.out.println("배열 길이 입력: ");
		int len = scan.nextInt();
		// int arr[5]; <- c언어에선 이렇게 해야하는데 java는 나중에 해줘도됨 동적인 특징.
		int arr[] = new int[len]; // <- 자바는 이렇게 설정
		
		System.out.println("arr.length = "+arr.length);
		scan.close();
	}
	
	public static void main(String[] args) {
		new Ex02();
	}
}
